File: data/src/main/java/com/github/anrimian/musicplayer/data/database/mappers/CompositionCorruptionDetector.java
package com.github.anrimian.musicplayer.data.database.mappers;

import com.github.anrimian.musicplayer.data.storage.providers.music.StorageFullComposition;
import com.github.anrimian.musicplayer.domain.models.composition.CorruptionType;

public class CompositionCorruptionDetector {

    public static CorruptionType getCorruptionType(StorageFullComposition composition) {
        CorruptionType corruptionType = null;
        if (composition.getDuration() == 0) {
            corruptionType = CorruptionType.UNKNOWN;
        }
        return corruptionType;
    }
}

File: data/src/main/java/com/github/anrimian/musicplayer/data/database/mappers/CompositionCorruptionDetector.kt
package com.github.anrimian.musicplayer.data.database.mappers

import com.github.anrimian.musicplayer.data.storage.providers.music.StorageAudioFile
import com.github.anrimian.musicplayer.domain.models.composition.CorruptionType

object CompositionCorruptionDetector {

    fun getCorruptionType(composition: StorageAudioFile): CorruptionType? {
        return getCorruptionType(composition.duration)
    }

    fun getCorruptionType(duration: Long): CorruptionType? {
        return if (duration == 0L) {
            CorruptionType.UNKNOWN
        } else {
            null
        }
    }

}
